pkg/host/machine_info_linux.go

```diff
@@ -143,13 +143,64 @@ func getModulesInfo() ([]KernelModule, error) {
 }
 
 func getGlobsInfo(globs map[string]bool) (map[string][]string, error) {
+	var err error
 	files := make(map[string][]string, len(globs))
 	for glob := range globs {
-		matches, err := filepath.Glob(glob)
+		var (
+			addglobs []string
+			subglobs []string
+			matches  []string
+		)
+		tokens := strings.Split(glob, ":")
+		for _, tok := range tokens {
+			if strings.HasPrefix(tok, "-") {
+				subglobs = append(subglobs, tok[1:])
+			} else {
+				addglobs = append(addglobs, tok)
+			}
+		}
+		for _, g := range addglobs {
+			m, err := filepath.Glob(g)
+			if err != nil {
+				return nil, err
+			}
+			matches = append(matches, m...)
+		}
+		files[glob], err = excludeGlobs(removeDupValues(matches), subglobs)
 		if err != nil {
 			return nil, err
 		}
-		files[glob] = matches
 	}
 	return files, nil
 }
+
+func excludeGlobs(items, excludes []string) ([]string, error) {
+	var results []string
+	for _, exclude := range excludes {
+		matches, err := filepath.Glob(exclude)
+		if err != nil {
+			return nil, err
+		}
+		for _, m := range matches {
+			for _, item := range items {
```

Comment:
This is O(N^2) complexity and it duplicates every non-excluded file N times in results and I don't think it's even working, is it? If matches=["a", "b"], on the first iteration we will append "b" to results and on the second iteration we will append "a" and as the result we did not exclude anything at all...
It will work faster (O(N)) and correctly if we put all exclude matches into a map first and then do 1 pass over items and remove excluded items.
And please add tests. This is easily unit-testable code.

Reply:
It works acutally because items being assigned by results
items = results

But let me rework to fit for O(N) too.